tests/test_equilibrium.py

```diff
@@ -464,3 +465,141 @@ def test_backward_compatible_load_and_resolve():
+    https://arxiv.org/pdf/1207.4419.pdf  (Table 3.5)
+    """
+    eq = Equilibrium.load(".//tests//inputs//low-beta-shifted-circle.h5")
+
+    eq_keys = ["iota", "iota_r", "a", "rho", "psi"]
+
+    psi = 0.25  # rho^2 (or normalized psi)
+    alpha = 0
+
+    eq_keys = ["iota", "iota_r", "a", "rho", "psi"]
+
+    data_eq = eq.compute(eq_keys)
+
+    iotas = np.interp(np.sqrt(psi), data_eq["rho"], data_eq["iota"])
+    shears = np.interp(np.sqrt(psi), data_eq["rho"], data_eq["iota_r"])
+
+    N = int((2 * eq.M_grid) * 4 + 1)
+
+    zeta = np.linspace(-1.0 * np.pi / iotas, 1.0 * np.pi / iotas, N)
+    theta_PEST = alpha * np.ones(N, dtype=int) + iotas * zeta
+
+    coords1 = np.zeros((N, 3))
+    coords1[:, 0] = np.sqrt(psi) * np.ones(N, dtype=int)
+    coords1[:, 1] = theta_PEST
+    coords1[:, 2] = zeta
+
+    # Creating a grid along a field line
+    c1 = eq.compute_theta_coords(coords1)
+    grid = Grid(c1, sort=False)
+
+    data_keys = [
+        "kappa",
+        "|grad(psi)|^2",
+        "grad(|B|)",
+        "grad(alpha)",
+        "grad(psi)",
+        "B",
+        "grad(|B|)",
+        "iota",
+        "|B|",
+        "B^zeta",
+        "cvdrift0",
+        "cvdrift",
+        "gbdrift",
+    ]
+
+    data = eq.compute(data_keys, grid=grid, override_grid=False)
+
+    psib = data_eq["psi"][-1]
+
+    # signs
+    sign_psi = psib / np.abs(psib)
+    sign_iota = iotas / np.abs(iotas)
+
+    # normalizations
+    Lref = data_eq["a"]
+    Bref = 2 * np.abs(psib) / Lref**2
+
+    modB = data["|B|"]
+    bmag = modB / Bref
+
+    x = Lref * np.sqrt(psi)
+    s_hat = -x / iotas * shears / Lref
+
+    grad_psi = data["grad(psi)"]
+    grad_alpha = data["grad(alpha)"]
+
+    iota = data["iota"]
+
+    gradpar = Lref * data["B^zeta"] / modB
+
+    gds21 = -sign_iota * np.array(dot(grad_psi, grad_alpha)) * s_hat / Bref
+
+    gbdrift = np.array(dot(cross(data["B"], data["grad(|B|)"]), grad_alpha))
+    gbdrift *= -sign_psi * 2 * Bref * Lref**2 / modB**3 * np.sqrt(psi)
+
+    cvdrift = (
+        -sign_psi
+        * 2
+        * Bref
+        * Lref**2
+        * np.sqrt(psi)
+        * dot(cross(data["B"], data["kappa"]), grad_alpha)
+        / modB**2
+    )
+
+    cvdrift0 = np.array(dot(cross(data["B"], data["grad(|B|)"]), grad_psi))
+    cvdrift0 *= sign_iota * sign_psi * s_hat * 2 / modB**3 / np.sqrt(psi)
+
+    ## Comparing coefficient calculation here with coefficients from compute/_mtric
+    cvdrift_2 = -2 * sign_psi * Bref * Lref**2 * np.sqrt(psi) * data["cvdrift"]
+    gbdrift_2 = -2 * sign_psi * Bref * Lref**2 * np.sqrt(psi) * data["gbdrift"]
+
+    # The error here should be of the same order as the max force error
+    np.testing.assert_allclose(gbdrift, gbdrift_2, atol=1e-5, rtol=1e-5)
+    np.testing.assert_allclose(cvdrift, cvdrift_2, atol=8e-4, rtol=9e-5)
+
+    a0_over_R0 = Lref * np.sqrt(psi)
+
+    # For the rest of the expressions, the error ~ a0_over_R0
+    fudge_factor1 = -3.8
```

Comment:
what is the fudge factor for?

Reply:
Because the real factor is a combination of iota, R0, a0, B0, pi, sqrt(2) and several such variables.

Note that bounce averaging operator is not going to include any of the geometric coefficients with these fudge factors. These factors are only used for the analytical geometric coefficients. Note that all fudge factors are actually scalars.

Let's say all the analytical geometric coefficients are in a vector X and their actual numerical values are in a vector Y.

The test is that
the numerical bounce-averaged drift  = f(Y, pitch_angle) 
evaluated numerically using the bounce-averaging PR must match 
the analytical bounce-averaged drift  = X[0] * E(pitch_angle) + X[1]* K(pitch_angle) 
where E and K are elliptic integrals. 

So we will chose a fixed set of pitch_angles and calculate the numerical solution and compare it with the analytical solution. The fudge factors make sure that the geometric coefficients are close enough (with a few percent) so that if the bounce-averaged drifts differ a lot, then something is wrong in a bounce-average PR.

Does that make sense?

Reply:
Ah so these are from the thesis, like a fit to make the analytic expression match numerics? Or are they just so that the magnitudes are close, then the test is really to ensure that the trends of the analytic expressions match the numerical values (which are the "ground truth")?